ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWT.scala

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.ledger.api.auth
+
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.jwt.JwtVerifier
+import com.digitalasset.ledger.api.auth.AuthServiceJWT.Error
+import io.grpc.Metadata
+import org.slf4j.{Logger, LoggerFactory}
+import spray.json._
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/** The JWT token payload used in [[AuthServiceJWT]]
+  *
+  * For forward/backward compatibility reasons, all fields are optional.
+  *
+  * The information in this token may be used to fill in missing data in gRPC ledger API requests.
+  */
+case class AuthServiceJWTPayload(
+    ledgerId: Option[String],
+    participantId: Option[String],
+    applicationId: Option[String],
+    exp: Option[String],
+    admin: Option[Boolean],
+    actAs: Option[List[String]],
+    readAs: Option[List[String]]
+)
+
+object AuthServiceJWTProtocol extends DefaultJsonProtocol {
+  implicit val jwtPayloadFormat = jsonFormat7(AuthServiceJWTPayload.apply)
+}
+
+/** An AuthService that reads a JWT token from a `Authorization: Bearer` HTTP header.
+  * The token is expected to use the format as defined in [[AuthServiceJWTPayload]]:
+  */
+class AuthServiceJWT(verifier: JwtVerifier) extends AuthService {
+
+  protected val logger: Logger = LoggerFactory.getLogger(AuthServiceJWT.getClass)
+
+  override def decodeMetadata(headers: Metadata): CompletionStage[Claims] = {
+    decodeAndParse(headers).fold(
+      error => {
+        logger.warn("Authorization error:", error)
+        CompletableFuture.completedFuture(Claims.empty)
+      },
+      token => CompletableFuture.completedFuture(payloadToClaims(token))
+    )
+  }
+
+  private[this] def parsePayload(jwtPayload: String): Either[Error, AuthServiceJWTPayload] = {
+    import AuthServiceJWTProtocol._
+    Try(JsonParser(jwtPayload).convertTo[AuthServiceJWTPayload]).toEither.left.map(t =>
+      Error("Could not parse JWT token: " + t.getMessage))
+  }
+
+  private[this] def decodeAndParse(headers: Metadata): Either[Error, AuthServiceJWTPayload] = {
+    val bearerTokenRegex = "[Bb]earer: (.*)".r
```

Comment:
```suggestion
    val bearerTokenRegex = "[Bb]earer (.*)".r
```
There shouldn't be a `:`, and maybe even forcing the capital `B` should be okay. See https://tools.ietf.org/html/rfc6750#section-2.1